qiskit_ionq/helpers.py

```diff
@@ -448,6 +448,20 @@ def get_user_agent():
     )
 
 
+class SafeEncoder(json.JSONEncoder):
+    """
+    Custom JSON encoder that handles non-JSON-safe objects by converting them to strings.
+    """
+
+    def default(self, o):
+        try:
+            return super().default(o)
+        except TypeError:
+            return (
+                str(o) if "__str__" in vars(type(o)) else repr(o)
```

Comment:
would prefer another try catch here, maybe you can set it up as a for loop with a few options for the source of the string?

something like

funcs = [ lambda o: super().default(o), ...]

for func in funcs:
   try:
      return func()
   except:
       log
return 'unknown'

Reply:
great yeah i updated to do that - ill look into adding the field name now